Test failure due to missing symbol `_ZN7edm4hep17MutableMCParticle11setMomentumENS_8Vector3fE` on LCG_101 ubuntu

- Goal: I'm attempting to compile EDM4hep in LCG with a more recent version of podio than is present in LCG_101
 - OS version: LCG_101/x86_64-ubuntu2004-gcc9-opt
 - Compiler version: gcc9
 - Package version: ffcdb0b34766e444d145367966037852f7692431
 - Reproduced by (originally inside a GitHub action with [run-lcg-views](https://github.com/AIDAsoft/run-lcg-views)):
 **Note:** with `CMAKE_PREFIX_PATH` unset as per #166, and therefore without HepPDT.
```console
#    - name: Build in LCG view
#      uses: aidasoft/run-lcg-view@v1
#      with:
#        release-platform: ${{ matrix.LCG }}
#        run: |

docker run --rm -it -v /cvmfs:/cvmfs:shared ghcr.io/aidasoft/ubuntu2004:latest /bin/bash
source /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/setup.sh
git clone https://github.com/AIDAsoft/podio
git clone https://github.com/key4hep/edm4hep

          # build podio
          cmake -B podio-build -S podio \
            -DCMAKE_INSTALL_PREFIX=podio-install \
            -DUSE_EXTERNAL_CATCH2=OFF
          make -C podio-build -j6 install
          
          # build EDM4HEP
          export PODIO=$PWD/podio-install
          unset CMAKE_PREFIX_PATH
          cmake -B edm4hep-build -S edm4hep \
            -DCMAKE_INSTALL_PREFIX=edm4hep-install \
            -DUSE_EXTERNAL_CATCH2=OFF
          make -C edm4hep-build -j6 install
```

This produces (reproducibly) the following output, after mostly uneventful compilation:
```console
[ 91%] Built target write_events
[ 92%] Built target reco_particle_ref
[ 93%] Linking CXX executable unittests
[ 94%] Built target read_events
[ 96%] Built target edm4hepRDF
[ 98%] Built target edm4hepDict
make[2]: Entering directory '/edm4hep-build'
make[2]: Leaving directory '/edm4hep-build'
make[2]: Entering directory '/edm4hep-build'
[ 98%] Building CXX object test/CMakeFiles/test_rdf.dir/test_rdf.cc.o
/edm4hep-build/test/utils/unittests: symbol lookup error: /edm4hep-build/test/utils/unittests: undefined symbol: _ZN7edm4hep17MutableMCParticle11setMomentumENS_8Vector3fE
CMake Error at /edm4hep-build/_deps/catch2-src/extras/CatchAddTests.cmake:45 (message):
  Error running test executable '/edm4hep-build/test/utils/unittests':

    Result: 127
    Output: 



make[2]: *** [test/utils/CMakeFiles/unittests.dir/build.make:104: test/utils/unittests] Error 1
make[2]: *** Deleting file 'test/utils/unittests'
make[2]: Leaving directory '/edm4hep-build'
make[1]: *** [CMakeFiles/Makefile2:1181: test/utils/CMakeFiles/unittests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

A potential hint may be in some cling ABI warnings (likely a different version of `libstdc++` between compilation time on ubunt 20.04.0 and current ubuntu 20.04.1 compiler, gcc 9.4.0-1ubuntu1~20.04.1):
```
Warning in cling::IncrementalParser::CheckABICompatibility():
  Possible C++ standard library mismatch, compiled with __GLIBCXX__ '20200808'
  Extraction of runtime standard library version was: '20210601'
```
though I have not seen this result in undefined symbols.

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -4,7 +4,7 @@ ENDIF()
 
 function(set_test_env _testname)
   set_property(TEST ${_testname} APPEND PROPERTY ENVIRONMENT
-      LD_LIBRARY_PATH=$<TARGET_FILE_DIR:edm4hep>:$<TARGET_FILE_DIR:podio::podio>:$ENV{LD_LIBRARY_PATH}
+      LD_LIBRARY_PATH=$<TARGET_FILE_DIR:edm4hep>:$<TARGET_FILE_DIR:podio::podio>:$<$<TARGET_EXISTS:edm4hepRDF>:$<TARGET_FILE_DIR:edm4hepRDF>:>$ENV{LD_LIBRARY_PATH}
       ROOT_INCLUDE_PATH=${PROJECT_SOURCE_DIR}/edm4hep:$ENV{ROOT_INCLUDE_PATH}
   )
   set_tests_properties(${_testname} PROPERTIES

